src/compile/mark/valueref.ts

```diff
@@ -105,29 +105,11 @@ export function bandRef(scaleName: string, band: number|boolean = true): VgValue
 }
 
 /**
- * Signal that returns the middle of a bin. Should only be used with x and y.
+ * Signal that returns the middle of a bin from start and end field. Should only be used with x and y.
  */
-function internalBinMidSignal(fieldDef: FieldDef<string>, scaleName: string) {
+function binMidSignal(start: string, end: string, scaleName: string) {
```

Comment:
I like my earlier suggestion better that the refactor you do here. 

`binMidSignal` should take `fieldDef` and optional `fieldDef2` (when something is optional, add `?` add make it the last function parameter) and then call `vgField` inside the method instead of requiring all of its call repeatedly calling `vgField` for both parameters.